### PR TITLE
fix(std/http): parsing of HTTP version header

### PR DESCRIPTION
**Pull request by Noxazer-FR**
The function parseHTTPVersion returns major variable first.
```Javascript
      return [major, minor];
    }
```
This is an misstyping available since the firsts versions of Deno.
Surely a problem when writing this line of code, that's not a big issue but it's better to fix it for further implementations.

I work actually on HTTP protocol implementation, with deno_std and based on rfc standards.
I've seen your message on deno_website repository, i have similar requirements for others projects so i work to create an modular implementation in full TypeScript, until Deno exist, it was not possible in securely way and both native V8/TS runtime.

That's the reason why i didn't used TypeScript until now, i seen the constraints of Node project too.
So thanks to Deno project authors.

### CODE REVIEW

File: std/http/_io.ts [hunk 1]
```diff
@@ -346,7 +346,7 @@ export async function readRequest(
   req.conn = conn;
   req.r = bufr;
   [req.method, req.url, req.proto] = firstLine.split(" ", 3);
-  [req.protoMinor, req.protoMajor] = parseHTTPVersion(req.proto);
+  [req.protoMajor, req.protoMinor] = parseHTTPVersion(req.proto);
```
Comment (bartlomieju) [inline std/http/_io.ts:349]: Yikes, that's a pretty bad typo, @Noxazer-FR can you add some test case ensuring we don't regress?

**Review by Noxazer-FR — COMMENTED** [review]
I prefer create a basic test for further versions, just i introduce the "version" variable to enable checking match between ServerRequest and parseHTTPVersion.
Testing req.protoMajor, req.protoMinor variables.

File: std/http/_io.ts (hunk 1 above)
Comment (Noxazer-FR) [reply std/http/_io.ts:349]: Yes of course, don't hesitate to leave comment on my work if necessary.

**Comment by Noxazer-FR** [comment]
Huuh i'm sorry but i don't understand what happened with theses tests. I tried to fix somes errors.
Can i get some explanations to fix the process ?

Thanks to you.

**Comment by bartlomieju** [comment]
@Noxazer-FR it looks like formatting/linting fails, make sure to run `tools/format.js` and `tools/lint.js`

**Comment by Noxazer-FR** [comment]
> 
> 
> @Noxazer-FR it looks like formatting/linting fails, make sure to run `tools/format.js` and `tools/lint.js`

Okay i'll look at this, i didn't have tools installed. After i'll check this error.
Regards.

**Comment by bartlomieju** [comment]
> > @Noxazer-FR it looks like formatting/linting fails, make sure to run `tools/format.js` and `tools/lint.js`
> 
> Okay i'll look at this, i didn't have tools installed. After i'll check this error.
> Regards.

@Noxazer-FR you don't need to install additional tools, all tooling required is in `third_party` submodule, the scripts themselves require `deno` to run.

**Comment by Noxazer-FR** [comment]
@bartlomieju I don't understand why all commands fails, to create this pull request i did it over github only.
And now i pull'd both repos from denoland and my fork, and finally i tried to run tests within powershell & wsl bash.
I have output:
```node
deno run --unstable --allow-write --allow-read --allow-run ./tools/format.js --check
dprint
rustfmt 123 file(s)
error: Uncaught (in promise) NotFound: No such file or directory (os error 2)
    at processResponse (deno:core/core.js:223:11)
    at Object.jsonOpSync (deno:core/core.js:246:12)
    at opRun (deno:runtime/js/40_process.js:19:17)
```
Up side is powershell, lower side is WSL.
Left side is deno-master, right side is my fork.
Screenshot_1
I apologize for my lack of understanding, regards.

**Comment by bartlomieju** [comment]
@Noxazer-FR looks like it fails on `rustfmt`, make sure that it's installed on your system. If you're having problems still I can reformat the code for you.

**Comment by Noxazer-FR** [comment]
Effectively i dont restarted my terminal/programs so, no environments variables was found for Cargo...
I tried to formater again.
```node
deno run --unstable --allow-write --allow-read --allow-run ./tools/format.js --check
dprint
rustfmt 123 file(s)
run tools/format.js
 M std/http/_io_test.ts
```
Finally i commited all changes. Just to be sure, i tried to run this both command:
```node
cargo test js_unit_tests
cargo -p deno --test integration_tests
---- js_unit_tests stdout ----
test_server starting...
target_dir P:\Users\Nox\Desktop\deno-nox\target\debug
thread 'js_unit_tests' panicked at 'failed to execute test_server: Os { code: 2, kind: NotFound, message: "Le fichier spécifié est introuvable." }', test_util\src\lib.rs:866:10
stack backtrace:
   0:     0x7ff67e31f309 - std::backtrace_rs::backtrace::dbghelp::trace
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4\/library\std\src\..\..\backtrace\src\backtrace\dbghelp.rs:98
   1:     0x7ff67e31f309 - std::backtrace_rs::backtrace::trace_unsynchronized
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4\/library\std\src\..\..\backtrace\src\backtrace\mod.rs:66
   2:     0x7ff67e31f309 - std::sys_common::backtrace::_print_fmt
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4\/library\std\src\sys_common\backtrace.rs:79
   3:     0x7ff67e31f309 - std::sys_common::backtrace::_print::{{impl}}::fmt
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4\/library\std\src\sys_common\backtrace.rs:58
   4:     0x7ff67e33e9eb - core::fmt::write
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4\/library\core\src\fmt\mod.rs:1080
   5:     0x7ff67dd80998 - std::io::Write::write_fmt<test::helpers::sink::Sink>
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4\library\std\src\io\mod.rs:1516
   6:     0x7ff67e314cb1 - std::io::impls::{{impl}}::write_fmt<Write>
                               at /rustc/7eac88abb2e57e752f3302f02be5f3ce3d7adfb4\/library\std\src\io\impls.rs:179
```
I have a panicked output log yet, **it's actually not important for closing this pull request**_, but it will be useful to debug the variable (file) who's unable to be read or found.
I have to prepare an environment to check, and reproduce this error._

@bartlomieju Many thanks for your help, i discovered the process to make better pull request on the way!

**Comment by bartlomieju** [comment]
>I have a panicked output log yet, it's actually not important for closing this pull request, but it will be useful to debug the variable (file) who's unable to be read or found.
I have to prepare an environment to check, and reproduce this error.

@Noxazer-FR this problems is caused by `test_server` binary not being present in `target/debug/` directory. You can fix that by `cargo build --bin test_server`

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @Noxazer-FR

**Comment by Noxazer-FR** [comment]
Thanks you too @bartlomieju

**Pull request merged**